agent/api/task/task.go

```diff
@@ -798,11 +800,23 @@ func (task *Task) addFirelensContainerDependency() error {
 		return errors.New("unable to add firelens container dependency because there's no firelens container")
 	}
 
+	if firelensContainer.HasContainerDependencies() {
+		// If firelens container has any container dependency, we don't add internal container dependency that depends
+		// on it in order to be safe (otherwise we need to deal with circular dependency).
+		seelog.Warnf("Not adding container dependency to let firelens container %s start first, because it has dependency on other containers.", firelensContainer.Name)
+		return nil
+	}
+
 	for _, container := range task.Containers {
 		if container.DockerConfig.HostConfig == nil {
 			continue
 		}
 
+		// Firelens container itself could be using awsfirelens log driver. Don't add container dependency in this case.
+		if container.Name == firelensContainer.Name {
+			continue
+		}
+
 		hostConfig := &dockercontainer.HostConfig{}
 		err := json.Unmarshal([]byte(*container.DockerConfig.HostConfig), hostConfig)
 		if err != nil {
```

Comment:
I think we don't need the condition "!firelensContainer.DependsOnContainer(container.Name)" anymore, since you already check whether firelensContainer has any dependency beforehand. So if it already comes to this point, it should already be true.

Reply:
good catch, fixed.